src/libpython.cpp

```diff
@@ -53,14 +53,32 @@ void lastDLErrorMessage(std::string* pError)
 #endif
 }
 
+bool loadSymbol(void* pLib, const std::string& name, void** ppSymbol, std::string* pError);
+
 bool loadLibrary(const std::string& libPath, void** ppLib, std::string* pError)
 {
-  *ppLib = NULL;
 #ifdef _WIN32
+  // assume that the symbols are not in the current process
   *ppLib = (void*)::LoadLibraryEx(libPath.c_str(), NULL, 0);
 #else
-  *ppLib = ::dlopen(libPath.c_str(), RTLD_NOW|RTLD_GLOBAL);
+  int symbols_present = 0;
+  void* pLib = NULL;
+  // check if the current process contains python symbols
+  pLib = ::dlopen(NULL, RTLD_NOW|RTLD_GLOBAL);
+  loadSymbol(pLib, "Py_IsInitialized", (void**) &Py_IsInitialized, pError);
+  ::dlclose(pLib);
+
+  if (Py_IsInitialized != NULL) {
```

Comment:
Is there a way to do this for Windows as well?

Reply:
we might use [this](https://github.com/JuliaPy/PyCall.jl/blob/eec4a0e810b9f87234569d742f6664ce16ee134d/src/startup.jl#L21-L36) from PyCall of Julia. I don't have a proper windows machine to test it though. 

This PR is needed for me to use the latest conda python (an older version of conda python doesn't load symbols into the main process). I am not sure if python symbols will be loaded into the main process in Windows as well.